### ISSUE TITLE
Library prefixes should allow starting with `_`

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#684]
Related to #294 

```dart
import 'a.dart' as _i1;
```

violates library_prefixes with "Use `lowercase_with_underscores` when specifying a library prefix." Buuuut it looks good to me. ☹️

**Comment by srawlins** [comment]
The current real world use case is in [code_builder](https://github.com/dart-lang/code_builder/blob/master/lib/src/scope.dart#L105).

**Comment by srawlins** [comment]
CC @matanlurey

**Comment by pq** [comment]
@munificent : this looks reasonable to me; curious if you have any thoughts?

**Comment by munificent** [comment]
It doesn't violate the style guide, so it's fine with me. It is a little weird to use a leading underscore for an identifier where privacy rules don't come into play. 😕

**Comment by pq** [comment]
Thanks @munificent! 

> It is a little weird to use a leading underscore for an identifier where privacy rules don't come into play.

I hear you.  I think the idea is to communicate a "this is an internal detail" intent (and avoid possible collisions).  @matanlurey and @srawlins can edify though.

**Comment by natebosch** [comment]
What will it take to change this? Looks like we need to tweak the regex here?
https://github.com/dart-lang/sdk/blob/1d7d954bf86b7d2cc49faeb016ddb0aca72cf46d/pkg/analyzer/lib/src/lint/util.dart#L21

**Comment by pq** [comment]
I'll take a look at this today. 👍

**Comment by bwilkerson** [comment]
Possibly. We'd need to see what other rules that change would effect, though. We might need to add a second regex that allows underscore and fix the code so that the new regex is used where appropriate.

**Comment by pq** [comment]
@bwilkerson : I've looked into it and it's not bad.  I've got an approach in mind.  Stay tuned! 😄

**Comment by natebosch** [comment]
Has this regressed? 2.0.0-dev.6.0 is reporting this lint for prefixes like `_1` or `_i1`

**Comment by pq** [comment]
Hey @natebosch!

It looks like in HEAD,`_i1` passes but `_1` does not.  That said, I'm not sure that `_1` was ever supported.  Could you confirm that `_i1` works for you?  Do you feel strongly that `_1` should pass?

**Comment by natebosch** [comment]
I *thought* I tested `_i1` and with  dev-3.0 it was still a lint.

Testing now with dev-6.0 I'm not seeing a lint. Either it's been fixed in between or I was holding it wrong when I was testing last night.

I think it's fine that `_1` produces a lint.

**Comment by pq** [comment]
Excellent.  Thanks for following up! 👍
